src/ompl/geometric/planners/prm/src/LazyPRM.cpp

```diff
@@ -467,25 +456,28 @@ void ompl::geometric::LazyPRM::markComponent(Vertex v, unsigned long int newComp
     }
 }
 
-long int ompl::geometric::LazyPRM::solutionComponent(std::pair<std::size_t, std::size_t> *startGoalPair) const
+long int ompl::geometric::LazyPRM::solutionComponent(Vertex &start, std::set<Vertex> &goals) const
 {
-    for (std::size_t startIndex = 0; startIndex < startM_.size(); ++startIndex)
+    goals.clear();
+
+    // TODO: Still need to handle multiple starts, will choose first start that works
+    for (const auto &s : startM_)
     {
-        long int startComponent = vertexComponentProperty_[startM_[startIndex]];
-        for (std::size_t goalIndex = 0; goalIndex < goalM_.size(); ++goalIndex)
+        auto startComponent = vertexComponentProperty_[s];
+        std::copy_if(std::begin(goalM_), std::end(goalM_), std::inserter(goals, std::begin(goals)),
+                     [&](const auto &g) { return vertexComponentProperty_[g] == startComponent; });
```

Comment:
The code does more work than necessary, since you can stop as soon as you find a goal state that is in the same component as the start state. You don't have to iterate over all goal states. No need to change, since optimizing this part of the code is probably not worth it.